### ISSUE TITLE
Fails compressing a JPG file

### ISSUE THREAD

**Issue by Zeokat** [dropbox/lepton#19]
I tryed the build provided for windows, but when i use a jpg it returns an error:
`C:\Users\VA\Downloads>lepton.exe 111.jpg 111.lep
lepton v1.0-`
`Input Size 90750 != Roundtrip Size 0
33448df365f5979a2e4d77ba40b6d4ff != 0600000005000000ffffffff00000000
ROUNDTRIP_FAILURE`

The image file is here (decompress it): [111.zip](https://github.com/dropbox/lepton/files/366904/111.zip)

Hope it helps improve the code.

**Comment by danielrh** [comment]
Thanks for the report! We'll look into it when we have a chance.
Do you have redistribution rights on the image? if so we can include it in the test suite to make sure no regression if we fix it

**Comment by Zeokat** [comment]
I have no redistribution rights on previous image...  but i noticed that any image that i save with Adobe Photoshop in JPG format and Quality 100, returns this error.

So here you have one image made by me that returns same error and with redistribution rights: 
[test.zip](https://github.com/dropbox/lepton/files/367217/test.zip)

**Comment by danielrh** [comment]
I think release 1.2 fixes the problem.
Reopen if the one without redistribution rights causes you issues still

**Comment by romiras** [comment]
I tried various JPG files with executable "lepton" and compression fails for me on any image.
Then I tried with "lepton-slow-best-ratio" and it works good, just failed once on very big image (~92MB) downloaded for test from Wikipedia . Other files were compressed without any errors.
If I compress with "lepton-slow-best-ratio" and then decompress with "lepton", I receive original file.

Tried on latest release 1.2.1 and on some commit made few days ago (don't remember which one) with same results for  executable "lepton".

$ ./lepton - < any_image.jpg > /dev/null
lepton v1.0-c378cbfa2daaa99e8828be7395013f94cedb1bcc
terminate called after throwing an instance of 'std::system_error'
  what():  Enable multithreading to use std::thread: Operation not permitted
SHORT_READ

File https://upload.wikimedia.org/wikipedia/commons/thumb/e/ec/Mona_Lisa%2C_by_Leonardo_da_Vinci%2C_from_C2RMF_retouched.jpg/300px-Mona_Lisa%2C_by_Leonardo_da_Vinci%2C_from_C2RMF_retouched.jpg
renamed to 3.jpg

$ ./lepton - < 3.jpg > /dev/null
lepton v1.0-c378cbfa2daaa99e8828be7395013f94cedb1bcc
terminate called after throwing an instance of 'std::system_error'
  what():  Enable multithreading to use std::thread: Operation not permitted
Input Size 35195 != Roundtrip Size 0
e2a450e1b96e4557e1b8527dfcc75a77 != 9e000000030000006a24eabf1824eabf
ROUNDTRIP_FAILURE
SHORT_READ

File https://upload.wikimedia.org/wikipedia/commons/e/ec/Mona_Lisa%2C_by_Leonardo_da_Vinci%2C_from_C2RMF_retouched.jpg  (~92MB) renamed to 4.jpg

$ ./lepton-slow-best-ratio - < 4.jpg > /dev/null
lepton v1.0-c378cbfa2daaa99e8828be7395013f94cedb1bcc
OOM
SHORT_READ


Tested on Ubuntu 14.04 LTS 32 bit, 8 GB RAM on-board with almost 4 GB free at time of execution.

$ uname -rmpsv
Linux 3.13.0-92-generic #139-Ubuntu SMP Tue Jun 28 20:42:32 UTC 2016 i686 i686

Processor info:
$ cat /proc/cpuinfo |less
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 60
model name      : Intel(R) Core(TM) i5-4690 CPU @ 3.50GHz
stepping        : 3
microcode       : 0x1a
cpu MHz         : 800.000
cache size      : 6144 KB
physical id     : 0
siblings        : 4
core id         : 0
cpu cores       : 4
apicid          : 0
initial apicid  : 0
fdiv_bug        : no
f00f_bug        : no
coma_bug        : no
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm ida arat xsaveopt pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm
bogomips        : 7000.25
clflush size    : 64
cache_alignment : 64
address sizes   : 39 bits physical, 48 bits virtual
power management:
